tests/generator/base.py

```diff
@@ -456,3 +456,29 @@ def assert_ovs(self, file_contents_map):
                 self.assertEqual(link_target,
                                  os.path.join(
                                     '/', 'run', 'systemd', 'system', fname))
+
+    def assert_sriov(self, file_contents_map):
+        systemd_dir = os.path.join(self.workdir.name, 'run', 'systemd', 'system')
+        if not file_contents_map:
+            # in this case we assume no SR-IOV configuration should be present
+            self.assertFalse(glob.glob(os.path.join(systemd_dir, '*netplan-sriov-*.service')))
+            return
+
+        self.assertEqual(set(os.listdir(self.workdir.name)) - {'lib'}, {'etc', 'run'})
+        sriov_systemd_dir = set(os.listdir(systemd_dir))
+        sriov_systemd_dir.remove('systemd-networkd.service.wants')
+        sriov_systemd_dir.remove('multi-user.target.wants')
+        sriov_systemd_dir.remove('netplan-ovs-cleanup.service')
+        self.assertEqual(sriov_systemd_dir, {'netplan-sriov-' + f for f in file_contents_map})
```

Comment:
suggestion (non-blocking): use the glob to produce the service file

(absolutely untested)
This should take care of the empty map for free, and voids the need to list all the extra service files.

```python
    def assert_sriov(self, file_contents_map):
        systemd_dir = os.path.join(self.workdir.name, 'run', 'systemd', 'system')
        sriov_systemd_dir = glob.glob(os.path.join(systemd_dir, '*netplan-sriov-*.service'))
        self.assertEqual(sriov_systemd_dir, {'netplan-sriov-' + f for f in file_contents_map})
        self.assertEqual(set(os.listdir(self.workdir.name)) - {'lib'}, {'etc', 'run'})
```

Reply:
Nice. Applied with small modifications.